src/scripts/pokeballs/Pokeballs.ts

```diff
@@ -74,12 +74,20 @@ class Pokeballs implements Feature {
             }, 1250, 'Increased catch rate on water routes', new RouteKillRequirement(10, GameConstants.Region.hoenn, 101)),
 
             new Pokeball(GameConstants.Pokeball.Lureball, () => {
-                const numLandPokemon = Routes.getRoute(player.region,player.route()).pokemon.land.length > 0;
-                const isWaterPokemon = Routes.getRoute(player.region,player.route()).pokemon.water.includes(Battle.enemyPokemon().name);
+                if (App.game.gameState == GameConstants.GameState.fighting && player.route()) {
+                    const numLandPokemon = Routes.getRoute(player.region,player.route()).pokemon.land.length > 0;
+                    const isWaterPokemon = Routes.getRoute(player.region,player.route()).pokemon.water.includes(Battle.enemyPokemon().name);
 
-                // If route has Land Pokémon and the current pokémon is a Water Pokémon
-                if (numLandPokemon == true && isWaterPokemon == true) {
-                    return 15;
+                    // If route has Land Pokémon and the current pokémon is a Water Pokémon
+                    if (numLandPokemon == true && isWaterPokemon == true) {
+                        return 15;
+                    }
+                }
+                if (App.game.gameState == GameConstants.GameState.dungeon) {
+                    // If player in a dungeon and the enemy pokémon is a water type
+                    if (Battle.enemyPokemon().type1 == PokemonType.Water || Battle.enemyPokemon().type2 == PokemonType.Water) {
```

Comment:
This dungeon code does not fit the description of the lure ball. It claims it helps on fished mons, but this is just water mons. Not sure if i am caught up in a weird detail, or we either need to remove this or change the description a bit

Reply:
Decided to scrap the dungeon portion after giving it some more thought. It doesn't fit canon and was a cheap way around dungeons not having "fishing" encounters like routes.